pilot/pkg/bootstrap/server.go

```diff
@@ -421,18 +422,13 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 	var clients []*mcpclient.Client
 	var conns []*grpc.ClientConn
 
-	for _, addr := range args.MCPServerAddrs {
-		u, err := url.Parse(addr)
-		if err != nil {
-			return err
-		}
-
+	for _, configSource := range s.mesh.ConfigSources {
 		securityOption := grpc.WithInsecure()
-		if u.Scheme == "mcps" {
+		if configSource.TlsSettings != nil && configSource.TlsSettings.Mode == istio_networking_v1alpha3.TLSSettings_ISTIO_MUTUAL {
 			requiredFiles := []string{
-				args.MCPCredentialOptions.CertificateFile,
-				args.MCPCredentialOptions.KeyFile,
-				args.MCPCredentialOptions.CACertificateFile,
+				configSource.TlsSettings.ClientCertificate,
+				configSource.TlsSettings.PrivateKey,
+				configSource.TlsSettings.CaCertificates,
```

Comment:
if its ISTIO mutual, you use certs from well defined paths (/etc/istio/... See networking/core/v1alpha3/clusters.go for details).

Reply:
Then how about these three `TlsSettings.ClientCertificate`, `TlsSettings.PrivateKey`, `TlsSettings.CaCertificates`? When should they be used?

Reply:
IF client specifies any thing other than ISTIO_MUTUAL as the TLS mode, then we use the fields specified..